string obj
---
bool success
int16 n
